go/client/chat_svc_handler.go

```diff
@@ -281,6 +282,39 @@ func (c *chatServiceHandler) SetUnfurlSettingsV1(ctx context.Context, opts setUn
 	return Reply{Result: true}
 }
 
+func (c *chatServiceHandler) GetDeviceInfoV1(ctx context.Context, opts getDeviceInfoOptionsV1) Reply {
+	client, err := GetUserClient(c.G())
+	if err != nil {
+		return c.errReply(err)
+	}
+	user, err := client.LoadUserByName(ctx, keybase1.LoadUserByNameArg{Username: opts.Username})
+	if err != nil {
+		return c.errReply(err)
+	}
+	arg := keybase1.LoadUserPlusKeysV2Arg{
+		Uid: user.Uid,
+		Oa:  keybase1.OfflineAvailability_BEST_EFFORT,
+	}
+	them, err := client.LoadUserPlusKeysV2(ctx, arg)
+	if err != nil {
+		return c.errReply(err)
+	}
+	var res chat1.GetDeviceInfoRes
+	for _, m := range them.Current.DeviceKeys {
+		dev := chat1.DeviceInfo{
+			DeviceID:          string(m.DeviceID),
+			DeviceDescription: m.DeviceDescription,
+			DeviceType:        m.DeviceType,
+			DeviceCtime:       m.Base.CTime.UnixSeconds(),
+		}
+		if string(m.DeviceID) != "" && m.Base.Revocation == nil {
```

Comment:
`UserPlusKeysV2` contains encryption keys that are not devices, and `Current` contains revoked devices.  Maybe there's a better way to remove those from this list rather than the string and nil compare?

Reply:
nil compare doesn't seem objectionable. Do you know what the non-device keys are?

Reply:
> Do you know what the non-device keys are?

I guess they are device keys: each device add adds two sigchain entries, one that says the device name and one that says "encryption keys for device name".  That latter one has no name or description or type, so it's ignored here.